esrally/driver/runner.py

```diff
@@ -61,16 +64,48 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         return False
 
 
-class DelegatingRunner(Runner):
+class SingleClusterDelegatingRunner(Runner):
+    def __init__(self, runnable, name, context_manager_enabled=False):
+        self.runnable = runnable
+        self.name = name
+        self.context_manager_enabled = context_manager_enabled
+
+    def __call__(self, *args):
+        # Single cluster mode: es parameter passed in runner is a client object for the "default" cluster
+        es = args[0]
+        return self.runnable(es['default'], *args[1:])
+
+    def __repr__(self, *args, **kwargs):
+        if self.context_manager_enabled:
+            return "user-defined context-manager enabled runner for [%s]" % self.name
+        else:
+            return "user-defined runner for [%s]" % self.name
+
+    def __enter__(self):
+        if self.context_manager_enabled:
+            self.runnable.__enter__()
+            return self
+        else:
+            return False
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if self.context_manager_enabled:
+            return self.runnable.__exit__(exc_type, exc_val, exc_tb)
+        else:
+            return False
+
+
+class MultiClusterDelegatingRunner(Runner):
```

Comment:
It seems `MultiClusterDelegatingRunner` does not support the context manager protocol.

Reply:
Right; I _thought_ we don't really need it as it is a special case after all. Then again, if it doesn't support the context manager protocol, I should clarify this in the docs, which sounds to me like more effort than adding the support :)